### DISCUSSION TITLE
Requests for new query functions

### DISCUSSION THREAD

**Discussion by lgy1027** [hyperledger/fabric-gateway#690] [Q&A]
Hello, I am new to fabric and now I need to operate the fabric network through the gateway. Currently, the data can be written normally and txHash is returned. However, I want to query the transaction details and block height based on TXhash. What method should be called for this?

**Comment by bestbeforetoday** [comment]
If you use the submit-async transaction flow in the API, the commit status returned to you includes the block number in which the transaction committed. The block number equates to the ledger height. You can use block eventing (with a start block number) to retrieve a specific block. There is an example of submit-async in the asset-transfer-basic sample ([Node](https://github.com/hyperledger/fabric-samples/blob/a5eea8cd5cceea088eea721902d4a54a76c4b869/asset-transfer-basic/application-gateway-typescript/src/app.ts#L173-L190), [Go](https://github.com/hyperledger/fabric-samples/blob/a5eea8cd5cceea088eea721902d4a54a76c4b869/asset-transfer-basic/application-gateway-go/assetTransfer.go#L218-L236), [Java](https://github.com/hyperledger/fabric-samples/blob/a5eea8cd5cceea088eea721902d4a54a76c4b869/asset-transfer-basic/application-gateway-java/src/main/java/App.java#L185-L207)).

Alternatively, you can invoke either the `GetBlockByTxID` or `GetTransactionByID` transaction function on the `qscc` system chaincode, which is available on all network peers. Both of these transaction functions take a channel name and a transaction ID as their two arguments. The return value will be a serialized protocol buffer message, which can be deserialized using the [fabric-protos](https://hyperledger.github.io/fabric-protos/) package.

The [off_chain_data](https://github.com/hyperledger/fabric-samples/tree/main/off_chain_data) sample provides an example of how to deserialize most of the block structure, including embedded transactions.
